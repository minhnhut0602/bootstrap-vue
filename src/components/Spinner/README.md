# Spinner

```vue live
<Spinner />
```

## Props

| Prop name | Description | Type | Values | Default  |
| --------- | ----------- | ---- | ------ | -------- |
| type      |             | Type | -      | 'border' |
