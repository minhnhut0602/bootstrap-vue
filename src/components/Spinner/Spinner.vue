<template>
    <div :class="classes" />
</template>

<script lang="ts">
import {PropType, computed, toRef} from 'vue';
import useSize, {sizeProps} from '../../composables/useSize';
import useVariant, {variantProps} from '../../composables/useVariant';
import useClasses from '../../composables/useClasses';
import useStringTemplate from '../../composables/useStringTemplate';

export const types = ['border', 'grow'] as const;

type Type = typeof types[number];
</script>

<script lang="ts" setup>
const props = defineProps({
    ...sizeProps,
    type: {
        type: String as PropType<Type>,
        default: 'border',
        validator: (t: Type) => types.includes(t),
    },
    ...variantProps,
});

const spinnerClass = useStringTemplate('spinner-{0}', toRef(props, 'type')).templatedString;

const {classes} = useClasses(computed(() => [
    useSize(props.size, `${spinnerClass.value}-{0}`).sizeClass.value,
    spinnerClass.value,
    useVariant(props.variant, 'text-{0}').variantClass.value,
]));
</script>

