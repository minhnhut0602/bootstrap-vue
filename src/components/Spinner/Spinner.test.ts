import Spinner from '.';

describe('template', () => {
    componentRenderTest(Spinner);

    componentWrapperClassTest(Spinner, {type: 'grow'}, 'spinner-grow');

    componentWrapperClassTest(Spinner, {size: 'lg'}, 'spinner-border-lg');

    componentWrapperClassTest(Spinner, {variant: 'danger'}, 'text-danger');
});
