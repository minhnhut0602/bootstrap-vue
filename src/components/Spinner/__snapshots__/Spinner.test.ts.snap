// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template renders default 1`] = `"<div class=\\"spinner-border-md spinner-border text-primary\\"></div>"`;
